tests/unit/test_collector.py

```diff
@@ -303,10 +306,34 @@ def test_31_ipmi_dcmi_collector_installed_and_okay(self):
         """Test ipmi dcmi collector can fetch correct number of metrics."""
         ipmi_dcmi_collector = IpmiDcmiCollector(Mock())
         ipmi_dcmi_collector.ipmi_dcmi = Mock()
+        ipmi_dcmi_collector.ipmi_tool = Mock()
+        ipmi_dcmi_collector.dmidecode = Mock()
+
+        mock_dcmi_payload = {"current_power": 105}
+
+        ipmi_dcmi_collector.ipmi_dcmi.get_current_power.return_value = mock_dcmi_payload
+        ipmi_dcmi_collector.ipmi_tool.get_ps_redundancy.return_value = (True, True)
+        ipmi_dcmi_collector.dmidecode.get_power_capacities.return_value = [1000, 1000]
+
+        payloads = ipmi_dcmi_collector.collect()
+
+        available_metrics = [spec.name for spec in ipmi_dcmi_collector.specifications]
+        self.assertEqual(len(list(payloads)), len(available_metrics))
+        for payload in payloads:
+            self.assertIn(payload.name, available_metrics)
+
+    def test_32_ipmi_dcmi_collector_get_ps_redundancy_not_ok(self):
+        """Test ipmi dcmi collector can fetch correct number of metrics."""
+        ipmi_dcmi_collector = IpmiDcmiCollector(Mock())
+        ipmi_dcmi_collector.ipmi_dcmi = Mock()
+        ipmi_dcmi_collector.ipmi_tool = Mock()
+        ipmi_dcmi_collector.dmidecode = Mock()
 
         mock_dcmi_payload = {"current_power": 105}
 
         ipmi_dcmi_collector.ipmi_dcmi.get_current_power.return_value = mock_dcmi_payload
+        ipmi_dcmi_collector.ipmi_tool.get_ps_redundancy.return_value = (False, True)
```

Comment:
It's okay to test with (False, True), However, if I peek into the function, it's impoosible to return (False, True).


Furthermore, do we want to test different combination to verify the fomula ? 
1. ps_redundancy  =   (True, True)  ,  (True, False) ,  (False, False)
2. get_power_capacities = [] ,  [1000], [1000, 1000] , [1000, 1500]

expected value of maximum_power_capacity and power_capacity_percentage